File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:

    int area(vector<int>& height, int left, int right){
        return min(height[left], height[right]) * (right - left);
    }

    int maxArea(vector<int>& height) {

        int left = 0;
        int right = height.size() - 1;

        int ans = 0;
        while(left < right){
            int currArea = area(height, left, right);
            ans = max(ans, currArea);

            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }

        return ans;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let l = 0;
    let r = height.length-1;
    let area = 0;
    while(l < r){
        let diff = r-l;
        let h = Math.min(height[l], height[r]);

        area = Math.max(diff * h, area);
        if(height[l] < height[r]){
            l++;
        }
        else r--;
    }    
    return area;
};